### PR TITLE
Separate handler filters in pkg/apiserver into pkg/apiserver/filters

### PR DESCRIPTION
**Pull request by sttts**
... in order to make dependencies explicit towards better composability.

---

This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/33478)

### CODE REVIEW

File: pkg/genericapiserver/filters/readonly.go [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+func IsReadOnlyReq(req http.Request) bool {
+	if req.Method == "GET" {
+		// TODO: add OPTIONS and HEAD if we ever support those.
+		return true
+	}
+	return false
+}
+
+// ReadOnly passes all GET requests on to handler, and returns an error on all other requests.
+func ReadOnly(handler http.Handler) http.Handler {
```
Comment (deads2k) [inline pkg/genericapiserver/filters/readonly.go]: I don't recognize this one.  Dead code?

**Comment by deads2k** [comment]
Assuming these are all straight moves, the new packages seem fine.

**Comment by deads2k** [comment]
Feel free to squash and tag on green.

File: pkg/genericapiserver/filters/readonly.go (hunk 1 above)
Comment (sttts) [inline pkg/genericapiserver/filters/readonly.go]: Yes, will remove it.

**Comment by sttts** [comment]
Rebased.

**Pull request merged**